### PR TITLE
Create project structure and netlist class

### PR DESCRIPTION
**Pull request by jameswolfeoliver**
Moves the reference genetic algorithm code to `python/ge_reference_code`

Creates project source code in `python/src/`

Created project tests code in `python/tests`

Creates netlist class, some mock file stuff and a simple test to make sure everything works

created pipenv for `python` folder

Tested using pycharm

### CODE REVIEW

File: python/src/file_io.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from abc import ABCMeta, abstractmethod
```
Comment (thrile) [inline python/src/file_io.py:1]: What is this going to be used for? Just general writing to files?

**Review by thrile — APPROVED** [review]
Looks good :)

**Review by mweerasi — APPROVED** [review]
Great work!

File: python/src/file_io.py (hunk 1 above)
Comment (jameswolfeoliver) [reply python/src/file_io.py:1]: Yup. Defining a File interface means we can make a mock version for tests so tests don't actually have to write to the file system

**Pull request merged**